src/Shop.java

```diff
@@ -0,0 +1,167 @@
+        if (choice.equals("p")) {
+            changeBike(idBike, "price");
+        }
+        if (choice.equals("n")) {
+            changeBike(idBike, "number");
+        }
+    }
+
+    private void changeBike(int idBike, String value) {
+        Bike bike = bikes.get(idBike);
+        if (value.equals("title")) {
+            System.out.println("Old title : " + bike.getTitle() + "\nEnter new title :");
+            bike.setTitle(UserInput.get());
+        }
+        if (value.equals("price")) {
+            System.out.println("Old price : " + bike.getPrice() + "\nEnter new price :");
+            bike.setPrice(Integer.parseInt(UserInput.get()));
+        }
+        if (value.equals("number")) {
+            System.out.println("Old number : " + bike.getNumber() + "\nEnter new number :");
+            bike.setNumber(Integer.parseInt(UserInput.get()));
+        }
+        bikes.set(idBike, bike);
+    }
+
+    private void addNewBike() {
+        int addNew = bikes.size();
+        bikes.add(new Bike("newBike", 0, 0));
+        changeBike(addNew, "title");
+        changeBike(addNew, "price");
+        changeBike(addNew, "number");
+        System.out.println("New Bike Created");
+    }
+
+    private void deleteOldBike(int id) {
+        int bikeSize = bikes.size();
+        if (id < bikeSize && bikeSize != 0) {
+            System.out.println("Old Bike : " + bikes.get(id).getTextBikePresentation());
+            System.out.println("Delet : Enter y or n");
+            String choice = UserInput.get();
+            if (choice.equals("y")) {
+                bikes.remove(id);
+            }
+            if (choice.equals("n")) {
+                return;
+            }
+        }
+        if (bikeSize == 0) {
+            System.out.println("Empty base");
+        }
+    }
+
+    private void getBikesFromTxt() {
+        File file = new File("BikeBase.txt");
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String line = null;
+            while ((line = bufferedReader.readLine()) != null) {
+                addBikeFromTxt(line);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File BikeBase.txt Not Found ");
+            e.printStackTrace();
+        } catch (IOException ex) {
+            System.out.println("File BikeBase.txt May be damaged");
+            ex.printStackTrace();
+        }
+    }
+
+    private void saveBikesInTxt() {
+        File file = new File("BikeBase.txt");
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file))) {
+            for (Bike bike : bikes) {
+                bufferedWriter.write(bike.getValueBikeInSaveTxt());
+                bufferedWriter.newLine();
+            }
+        } catch (IOException ex) {
+            System.out.println("File BikeBase.txt May be damaged");
+            ex.printStackTrace();
+        }
+    }
+
+    private void addBikeFromTxt(String bikeLine) {
```

Comment:
`addBike` is enough. Remove the rest if the name

Reply:
bikeLine -> bikeString

Reply:
ok